File: DSA_Pep/Basic/1.cpp
#include <bits/stdc++.h>
using namespace std;

// void sub(string a,string res=""){
//   if(res.length()==a.length()) {
//     cout<<res<<" ";
//     return;
//   }
//   sub(a.substr(1), res+=a[0]);
//   sub(a.substr(1), res);
// }

int main(){
  string a;
  cin>>a;
  for(int i=0;i<a.length();i++){
    string temp="";
    for(int j=i;j<a.length();j++){
      temp+= a[j];
      cout<<temp<<" ";
    }
  }
return 0;
}

File: competitive/HashCode/1.js
let n=3;
let arr =[1,2,3,4,5,6]
let i=1, j=n+1;
while(i<n && j<2*n){
  let temp =arr[i];
  arr[i]=arr[j]
  arr[j]=temp;
  i+=2;
  j++;
}
console.log(arr);